src/robotFactory.js

```diff
@@ -1,10 +1,96 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(y = 1) {
+    this.coords.y += y;
+  };
+
+  goBack(y = 1) {
+    this.coords.y -= y;
+  };
+
+  goLeft(x = 1) {
+    this.coords.x -= x;
+  };
+
+  goRight(x = 1) {
+    this.coords.x += x;
+  };
+
+  getInfo() {
+    return `Robot: ${this.name},`
+      + ` Chip version: ${this.chipVersion},`
+      + ` Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, { x, y, z = 0 }, chipVersion) {
+    super(
+      name,
+      weight, {
+        x,
+        y,
+        z,
+      }, chipVersion);
+
+    this.coords.z = z;
```

Comment:
since you do it anyway, it may look this way:
```suggestion
    this.coords.z = z || 0;
```
and no need to do destructuring assignment in the `constructor` arguments, and pass the new object in `super`

do the same in the classes above

Reply:
It'll be more readable

Reply:
Hi, thank you for check, but i already asked about this in chat and we decided to keep my option with Tetiana https://mate-parttime.slack.com/archives/C03UQS59NRY/p1674823760016929